labkey-client-api/src/org/labkey/remoteapi/domain/QueryFilter.java

```diff
@@ -0,0 +1,83 @@
+package org.labkey.remoteapi.domain;
+
+public class QueryFilter
+{
+    public enum Type
+    {
+        HAS_ANY_VALUE(""),
+
+        EQUAL("eq"),
+        DATE_EQUAL("dateeq"),
+
+        NEQ("neq"),
+        NOT_EQUAL("neq"),
+        DATE_NOT_EQUAL("dateneq"),
+
+        NEQ_OR_NULL("neqornull"),
+        NOT_EQUAL_OR_MISSING("neqornull"),
+
+        GT("gt"),
+        GREATER_THAN("gt"),
+        DATE_GREATER_THAN("dategt"),
+
+        LT("lt"),
+        LESS_THAN("lt"),
+        DATE_LESS_THAN("datelt"),
+
+        GTE("gte"),
+        GREATER_THAN_OR_EQUAL("gte"),
+        DATE_GREATER_THAN_OR_EQUAL("dategte"),
+
+        LTE("lte"),
+        LESS_THAN_OR_EQUAL("lte"),
+        DATE_LESS_THAN_OR_EQUAL("datelte"),
+
+        STARTS_WITH("startswith"),
+        DOES_NOT_START_WITH("doesnotstartwith"),
+
+        CONTAINS("contains"),
+        DOES_NOT_CONTAIN("doesnotcontain"),
+
+        CONTAINS_ONE_OF("containsoneof"),
+        CONTAINS_NONE_OF("containsnoneof"),
+
+        IN("in"),
+
+        EQUALS_ONE_OF("in"),
+
+        NOT_IN("notin"),
+        EQUALS_NONE_OF("notin"),
+
+        BETWEEN("between"),
+        NOT_BETWEEN("notbetween"),
+
+        IS_BLANK("isblank"),
+        IS_NOT_BLANK("isnonblank"),
+
+        MEMBER_OF("memberof");
+
+        public final String label;
+
+        Type(String label) {
```

Comment:
I don't think we need this class, look at the existing Filter class in remoteapi. That should get you pretty much everything you need, and the conditional format column name is always : format.column.

I also think we should handle the ConditionalFormat filters similar to how you did it for the Python API, where we have an array of filters. As an example replace the String setter/getter of queryFilter with:

setFilters(List<Filter> filters)

The only tricky part that the existing class doesn't handle is deserialize from JSON since we don't have that scenario currently. I don't think it would be difficult to create a Filter object from a URL filter expression but we can discuss and I'm happy to help with that impl.